дружественные числа/Program.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace дружественные_числа
+{
+    class Program
+    {
+        
+        static void Main(string[] args)
+        {
+            while(true){
+            Console.WriteLine("Введите значение, в пределах которого искать дружественные числа (Enter — отказ)");
+                var input = Console.ReadLine();
+
+                if (input.Length < 1)
+                    break;
+
+                if(!int.TryParse(input, out int n))
+                {
+                    Console.WriteLine("Ошибка ввода\n");
+                    continue;
+                }
+            for (int i = 0; i <= n; i++)
+            {
+                if (SummarizeDividers(SummarizeDividers(i)) == i && SummarizeDividers(i) != i)
+                    Console.WriteLine($"\n{i} и {SummarizeDividers(i)}\n");
```

Comment:
Чтобы не вычислять одно и тот же значение трижды ( а оно вычисляется циклом, количество вычислений пропорциональна значению числа) лучше использовать локальную переменную.
Кроме того, у вас каждая пара находится дважды. Лучше искать пару, чтобы одно число было меньше другого.
И не нужно так много пустых строк на выводе.
```suggestion
            {
                var s = SummarizeDividers(i);
                if (SummarizeDividers(s) == i && s > i)
                    Console.WriteLine($"{i} и {s}");
```